$DragonFly: src/secure/usr.sbin/sshd/Attic/loginrec.c.patch,v 1.2 2005/07/11 22:49:46 corecode Exp $
--- loginrec.c.orig	2005-07-11 11:14:15.000000000 +0200
+++ loginrec.c	2005-07-11 11:13:37.000000000 +0200
@@ -670,8 +670,8 @@
 	strncpy(ut->ut_name, li->username,
 	    MIN_SIZEOF(ut->ut_name, li->username));
 # ifdef HAVE_HOST_IN_UTMP
-	strncpy(ut->ut_host, li->hostname,
-	    MIN_SIZEOF(ut->ut_host, li->hostname));
+	realhostname_sa(ut->ut_host, sizeof ut->ut_host,
+	    &li->hostaddr.sa, li->hostaddr.sa.sa_len);
 # endif
 # ifdef HAVE_ADDR_IN_UTMP
 	/* this is just a 32-bit IP address */
